src/common/conf_file.cpp

```diff
@@ -540,6 +540,29 @@ int ConfFile::parse_uint8_vector(const char *val, size_t val_len, void *storage,
     return 0;
 }
 
+int ConfFile::parse_uint32_vector(const char *val, size_t val_len, void *storage,
+                                  size_t storage_len)
+{
+    assert(val);
+    assert(storage);
+    assert(val_len);
+
+    if (storage_len < sizeof(bool)) {
+        return -ENOBUFS;
+    }
+
+    char *filter_string = strndupa(val, val_len);
+    auto *target = (std::vector<uint32_t> *)storage;
+
+    char *token = strtok(filter_string, ",");
+    while (token != nullptr) {
+        target->push_back(atoi(token)); // int should be enough even on 32 bit systems
```

Comment:
you are parsing uint32_t here, but atoi returns signed int... there isn't enough room `INT32_MAX + 1` for that, so you should use atol, or strtou

```
$ $ cat /tmp/c.cpp 
#include <stdlib.h>
#include <stdio.h>
#include <inttypes.h>

int main(int argc, char *argv[])
{
	uint32_t u;

	u = atoi(argv[1]);
	printf("%" PRIu32 "\n", u);

	return 0;
}

$ g++ -m32 -o /tmp/c /tmp/c.cpp
$ /tmp/c 2147483648
2147483647

```

Reply:
It should be fine in that place because the MAVLink system ID is just 24 bit, but the method should be capable or uint32, if it is called that way.

Reply:
changed